reminders/scheduler.py

```diff
@@ -0,0 +1,338 @@
+        ))
+
+    @staticmethod
+    def __create_timer(functions, message, user_id):
+        """ create the timer object.
+            The scheduler needs a timer object that will go off when a spesific time has been reached
+            "Timer" is a class defined in pythons own threading library that get a tmme and a method to invoke.
+            This function set the current timer to be a new timer object with
+                time: the diffrence of the reminder time and the current time
+                target: the function given using the _get_function class
+                args: the argument that the function gets, a message and the user_id object
+
+            ** this function is run using a diffrent thread becuse the timer itself
+               can cause a few problems with the migrate and the current thread running it.
+        """
+
+        UserAlertScheduler.__current_timer = Timer(
+            UserAlertScheduler.__get_time_diffrence(UserAlertScheduler.__current_reminder.date_time),
+            UserAlertScheduler.__alert_user,
+            args=(functions, message, user_id)
+        )
+        UserAlertScheduler.__current_timer.start()  # start the timer object
+
+    @staticmethod
+    def __remove_alert(reminder_id):
+        """ remove the reminder with a reminder_id from the queue.
+            The scheudler will check if the remidner given is the current remidner
+            if so the scheudler will remove it and the current timer,
+            and will set a new remidner with its own timer according to the add_alert function
+        """
+
+        UserAlertScheduler.__logger.info("in remove alert")
+
+        if UserAlertScheduler.__current_reminder is not None:
+
+            if UserAlertScheduler.__current_reminder.id == reminder_id:
+
+                if UserAlertScheduler.__current_timer.is_alive():  # check if the timer is stil running
+                    UserAlertScheduler.__current_timer.cancel()
+                    UserAlertScheduler.__current_reminder = None
+                    UserAlertScheduler.__current_timer = None
+
+                    UserAlertScheduler.__logger.debug(
+                        f"removed the current reminder. last reminer: {Reminder.objects.get(id=reminder_id)}"
+                    )
+                else:
+                    UserAlertScheduler.__logger.warning(
+                        f"the remidner {UserAlertScheduler.__current_reminder} is the current remidner but has no timer"
+                    )
+
+    @staticmethod
+    def __modifie_alert(reminder):
+        """ modifie the reminder in the queue.
+            The scheudler will check if the given reminder is the current reminder object
+            if so the scheduler will chekc if the time has been change,
+            if the time has been incresed then the scheduler will try to add a new remidner insted
+            otherwise the reminder will be change in the scheduler to the reminder object given
```

Comment:
```suggestion
            if the time has been increased then the scheduler will try to add a new reminder instead
            otherwise, the remainder will be changed in the scheduler to the reminder object given
```

Reply:
done. thanks 😄